blogs/qserial.md

```diff
@@ -0,0 +1,243 @@
+Hey AngularJS devs!  Let's talk about executing async tasks serially with promises.
+
+First, we notice .then allows our success/fail callbacks to return a promise.  $q treats success/fail callbacks that return promises specially.  We'll look at two examples to see how $q treats success/fail functions that return promises differently from those that don't.
+<!--more-->
+
+Normal success function:
+<pre>
+$q.when()  
+.then(function () { console.log("foo"); })
+.then(function () { console.log("bar"); });
+</pre>
+<pre>
+foo
+bar
+</pre>
+<br/>
+Success function that returns a promise:
+<pre>
+$q.when()
+.then(function () { 
+  console.log("foo"); 
+  var bazDeferred = $q.defer();
+  var bazPromise = bazDeferred.promise;
+  return bazPromise;
+})
+.then(function () { console.log("bar"); });
+</pre>
+<pre>
+foo  
+</pre>
+<br/>
+As expected, our first example logs "foo" then "bar".  However, our second example only logs "foo".  $q does not continue executing the chain of promises until bazPromise has been resolved.  So "bar" is never logged.
+
+This is good!  $q allows us to execute chained promises sequentially by returning promises in our success functions.
+
+Let's see this $q feature in action:
+<pre>
+$q.when()
+.then(function () {
+  var deferred = $q.defer();
+  console.log("foo"); 
+  $timeout(function () { deferred.resolve(); }, 5000);
+  return deferred.promise;
+})
+.then(function () { console.log("bar"); });
+</pre>
+<pre>
+foo  
+//5 seconds
+bar
+</pre>
+<br/>
+Great, so now we see how we can execute promises sequentially.  How do we pass data through to the next promise?  Easy, we pass arguments to .resolve() or .reject().
+
+<pre>
+$q.when()
+.then(function () {
+  var deferred = $q.defer();
+  console.log("foo"); 
+  $timeout(function () { deferred.resolve("bar"); }, 5000);
```

Comment:
$timeout returns a promise, so it seems like this could also be

$timeout(5000).then(function () { deferred.resolve("bar"); })

Clunkier for sure but more "pure".